pkg/test/resourcefixture/testdata/basic/edgecontainer/v1beta1/edgecontainercluster/create.yaml

```diff
@@ -0,0 +1,35 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: edgecontainer.cnrm.cloud.google.com/v1beta1
+kind: EdgecontainerCluster
+metadata:
+  name: edgecontainercluster-${uniqueId}
+  labels:
+    label-one: "value-one"
+spec:
+  projectRef:
+    external: ${projectId}
+  location: us-central1
+  networking:
+    clusterIpv4CidrBlocks: ["10.0.0.0/16"]
+    servicesIpv4CidrBlocks: ["10.1.0.0/16"]
+  fleet:
+    projectRef: 
+      external: projects/${projectId}
```

Comment:
I am curious even I create a project as reference, I still cannot got the project number.
The project-${uniqueId} is the project ID but for project number I don't know if there is a good way to get it from the KKC reference. 
Not sure how I put the project number here
```
fleet:
    projectRef: 
      number: {}
```
Maybe I still need to go for option1 to have an external project number?

Reply:
When you reference to the dependency project by its id, KCC will use the number field of this dependency project as the value of fleet.project, because you configured [this](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/967/files#diff-2dd4cbc5f1a1f3b2a8660cab56b6d9466347533567f2d49f144fafe9c2fe5567R69-R70) in service mapping. You do not need to worry about the exact project number, the `name: project-${uniqueId}` does not mean we use `project-${uniqueId}` as value of fleet.project, it only helps KCC to find the your dependency project.
```
fleet:
    projectRef: 
      // Reference to the dependency project with id project-${uniqueId}, and we'll use it's project number
      name: project-${uniqueId}
```
Lmk if you still have questions.

Reply:
Gotcha, now I know how it works, the reference point to the project obj and apply the project number field https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_project_organization_policy
Which is defined in its TF template.